### PR TITLE
Add find by nric feature

### PR DESCRIPTION
**Pull request by yezkez10**
Added FindByNricCommand + its Parser
Added NricTest, passes.
Resolves issue https://github.com/AY2324S1-CS2103T-W09-3/tp/issues/80
Testcases from previously passes locally but not on build

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FindByNricCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+     * Parses the given {@code String} argument in the context of the FindByNricCommand
+     * and returns a FindByNricCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindByNricCommand parse(String args) throws ParseException {
+        String trimmedArg = args.trim();
+        if (trimmedArg.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindByNricCommand.MESSAGE_USAGE));
+        }
```
Comment (tohpinren) [inline src/main/java/seedu/address/logic/parser/FindByNricCommandParser.java:25]: Add checks for invalid nric inputs as well.
Right now `find_nric t1234567` `find_nric 1234567` `find_nric 1234567s` still works and finds 0 patients

Also add checks for starting letter to be S or T and 1234567 are numbers

Update tests accordingly

**Review by tohpinren — CHANGES_REQUESTED** [review]
Add parsing checks for invalid NRIC inputs so invalid inputs do not get executed. Add tests accordingly as well. Thanks!

File: src/test/java/seedu/address/model/person/NricContainsKeywordPredicateTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
```
Comment (Bryan-Goh) [inline src/test/java/seedu/address/model/person/NricContainsKeywordPredicateTest.java:6]: Maybe can add more logging in the test cases

**Review by Bryan-Goh — APPROVED** [review]
LGTM. Just a few nits to fix.

File: src/main/java/seedu/address/model/person/Nric.java [hunk 3]
```diff
@@ -10,8 +10,8 @@
 public class Nric {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "NRICs should contain 7 digits, with 2 capital letters at the beginning and end";
-    public static final String VALIDATION_REGEX = "\\p{Alpha}\\d{7}\\p{Alpha}";
+            "NRICs should contain 7 digits, with S or T at the beginning and a letter at the end";
+    public static final String VALIDATION_REGEX = "[ST][0-9]{7}[A-Za-z]";
```
Comment (tohpinren) [inline src/main/java/seedu/address/model/person/Nric.java]: Can add lower case s and t also hahhaa ps

**Review by tohpinren — CHANGES_REQUESTED** [review]
Add lowercase st

File: src/main/java/seedu/address/model/person/Nric.java (hunk 3 above)
Comment (yezkez10) [reply src/main/java/seedu/address/model/person/Nric.java]: done!

File: src/main/java/seedu/address/model/person/Nric.java [hunk 4]
```diff
@@ -11,7 +11,7 @@ public class Nric {
 
     public static final String MESSAGE_CONSTRAINTS =
             "NRICs should contain 7 digits, with S or T at the beginning and a letter at the end";
-    public static final String VALIDATION_REGEX = "[ST][0-9]{7}[A-Za-z]";
+    public static final String VALIDATION_REGEX = "[stST][0-9]{7}[A-Za-z]";
```
Comment (Bryan-Goh) [inline src/main/java/seedu/address/model/person/Nric.java:14]: the changes were done according to the request. LGTM

**Review by Bryan-Goh — APPROVED** [review]

**Review by simbayippy — APPROVED** [review]
LGTM

**Review by tohpinren — APPROVED** [review]

**Pull request merged**